controllers/storageoscluster/controller.go

```diff
@@ -118,8 +118,8 @@ func (c *StorageOSClusterController) UpdateStatus(ctx context.Context, obj clien
 	csiCondition := getCSICondition(ctx, c.Client, obj.GetNamespace(), log)
 	meta.SetStatusCondition(&cluster.Status.Conditions, csiCondition)
 
-	metricsEnabled, err := AreMetricsEnabled(cluster)
-	if err != nil || !metricsEnabled {
+	metricsEnabled := cluster.Spec.Metrics.Enabled
+	if !metricsEnabled {
```

Comment:
No need for that extra step/variable IMO
```suggestion
	if !cluster.Spec.Metrics.Enabled {
```